web/src/features/charts/MissingExchangeData.tsx

```diff
@@ -0,0 +1,46 @@
+import { useTranslation } from 'react-i18next';
+
+import { useNetExchangeChartData } from './hooks/useNetExchangeChartData';
+import { AreaGraphElement } from './types';
+
+const SIGNIFICANT_THRESHOLD = 0.2;
+
+const getTotal = (object: object) => Object.keys(object).length;
+const getNulls = (object: object) =>
+  Object.values(object).filter((v) => v === null).length;
+
+function getIsMissingSignificantExchangeData(chartData: AreaGraphElement[]) {
+  let nullCount = 0;
+  let total = 0;
+  for (const exchange of chartData.map(({ meta: { exchange } }) => exchange)) {
+    nullCount += getNulls(exchange);
+    total += getTotal(exchange);
+  }
+  return nullCount / total >= SIGNIFICANT_THRESHOLD;
+}
+
+export function MissingExchangeDataDisclaimer() {
+  const { data, isLoading, isError } = useNetExchangeChartData();
+  const { t } = useTranslation();
+
+  if (isLoading || isError || !data) {
+    return null;
+  }
+
+  const { chartData } = data;
+
+  const date = new Date();
+  date.setDate(date.getDate() - 1);
+  date.setHours(0, 0, 0, 0);
+
+  const isWithinPrevious48Hours = date < new Date(chartData.at(-1)!.datetime);
+  const isMissingSignificantExchangeData = getIsMissingSignificantExchangeData(chartData);
+
+  if (isWithinPrevious48Hours && isMissingSignificantExchangeData) {
+    return (
+      <p className="prose my-1 rounded bg-gray-200 p-2 text-xs leading-snug dark:bg-gray-800 dark:text-white dark:prose-a:text-white">
+        {t('country-history.exchange-delay')}
+      </p>
+    );
+  }
```

Comment:
It should probably return null here if this is false.

Undefined might work but I think the convention is that components should always return values (never undefined), but feel free to double check as im not 100% sure.